tools/livelog/writer/stream_handle.go

```diff
@@ -117,42 +112,20 @@ func (self *StreamHandle) WriteTo(target io.Writer) (n int64, err error) {
 	}
 
 	for event := range self.events {
-		pendingBuf := 0
-		self.pendingEvents[0] = event
-
-		// Build a list of all the pointers to the events we need to update. This
-		// has the very important effect of emptying the channel which may be
-		// building up very quickly.
```

Comment:
I guess this wasn't actually important?

Reply:
As I understood it, jlal saw that the channel's buffer was filling up and causing the producer to stop sending events to this handler.  So the solution was to add an additional buffer *after* the channel, of the same size.  The effect was to write a lot of confusing lines of code to double the channel buffer size.  I reduced this to the more efficient 

```diff
-const EVENT_BUFFER_SIZE = 100
+const EVENT_BUFFER_SIZE = 200
```